### DISCUSSION TITLE
Performing MD with 4GPU and 32 cores

### DISCUSSION THREAD

**Discussion by cui-hq** [deepmodeling/deepmd-kit#4451] [Q&A]
I want to perform MD with 4GPU and 32 cores,but when I perform MD using "mpirun -n 32 -i  input.lammps" , there is an ERROR, “There are not enough slots available in the system to satisfy the 6
slots that were requested by the application:

  lmp_mpi

Either request fewer slots for your application, or make more slots
available for use.

A "slot" is the Open MPI term for an allocatable unit where we can
launch a process.  The number of slots available are defined by the
environment in which Open MPI processes are run:

  1. Hostfile, via "slots=N" clauses (N defaults to number of
     processor cores if not provided)
  2. The --host command line parameter, via a ":N" suffix on the
     hostname (N defaults to 1 if not provided)
  3. Resource manager (e.g., SLURM, PBS/Torque, LSF, etc.)
  4. If none of a hostfile, the --host command line parameter, or an
     RM is present, Open MPI defaults to the number of processor cores

In all the above cases, if you want Open MPI to default to the number
of hardware threads instead of the number of processor cores, use the
--use-hwthread-cpus option.

Alternatively, you can use the --oversubscribe option to ignore the
number of available slots when deciding the number of processes to
launch.
--------------------------------------------------------------------------
”
I want to know how to perform MD with 4GPUs and 32cores

**Comment by njzjz** [comment]
The best practice is that the number of MPI ranks is no more than the number of GPUs.
